packages/discovery-provider/ddl/migrations/0093_splits_in_purchases.sql

```diff
@@ -0,0 +1,63 @@
+-- Recreates the extended purchase splits from the information in the purchases table
+-- Required to populate purchase details modal
+BEGIN;
+
+ALTER TABLE
+    usdc_purchases
+ADD
+    COLUMN IF NOT EXISTS splits JSONB;
+
+WITH relevant_timestamps AS (
+    SELECT
+        user_id,
+        MAX(block_timestamp) AS block_timestamp,
+        signature
+    FROM
+        user_payout_wallet_history
+        JOIN usdc_purchases ON seller_user_id = user_id
+    WHERE
+        block_timestamp < usdc_purchases.created_at
+    GROUP BY
+        user_id,
+        signature
+)
+UPDATE
+    usdc_purchases
+SET
+    splits = jsonb_build_array(
+        jsonb_build_object(
+            'user_id',
+            usdc_purchases.seller_user_id,
+            'amount',
+            usdc_purchases.amount,
+            'percentage',
+            100.0,
+            'eth_wallet',
+            users.wallet,
+            'payout_wallet',
+            COALESCE(
+                user_payout_wallet_history.spl_usdc_payout_wallet,
+                usdc_user_bank_accounts.bank_account
+            )
+        )
+    )
+FROM
+    usdc_purchases AS up
+    JOIN users ON users.user_id = up.seller_user_id
+    JOIN usdc_user_bank_accounts ON users.wallet = usdc_user_bank_accounts.ethereum_address
+    LEFT JOIN relevant_timestamps ON relevant_timestamps.user_id = users.user_id
+    AND relevant_timestamps.signature = up.signature
+    LEFT JOIN user_payout_wallet_history ON user_payout_wallet_history.user_id = users.user_id
+    AND user_payout_wallet_history.block_timestamp = relevant_timestamps.block_timestamp
+WHERE
+    up.splits IS NULL
+    AND up.signature = usdc_purchases.signature;
```

Comment:
isn't `up` the same as `usdc_purchases`?

Reply:
My understanding is that `up` is the `usdc_purchases` being joined, and the `usdc_purchases` itself is the context of the row being updated. In other words, we're updating `usdc_purchases` with values from `up`, and we need to make sure the `up` values are from the same row we're updating currently in `usdc_purchases`. It's kind of funky but I think it was necessary to use `usdc_purchases` in the other joins